pylibs/stress_tests/StressTestResult.py

```diff
@@ -0,0 +1,88 @@
+#!/usr/bin/env python3
+# Copyright (c) 2020, The OTNS Authors.
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+# 1. Redistributions of source code must retain the above copyright
+#    notice, this list of conditions and the following disclaimer.
+# 2. Redistributions in binary form must reproduce the above copyright
+#    notice, this list of conditions and the following disclaimer in the
+#    documentation and/or other materials provided with the distribution.
+# 3. Neither the name of the copyright holder nor the
+#    names of its contributors may be used to endorse or promote products
+#    derived from this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+
+import time
+
+from typing import Sequence, Any
+
+
+class StressTestResult(object):
+
+    def __init__(self, name: str, headers: Sequence[str]):
+        self.name = name
+        self.headers = tuple(headers)
+        self.rows = []
+        self._start_time = None
+        self._stop_time = None
+        self._failed = False
+        self._fail_msgs = []
+        self._fail_exception = None
+
+    @property
+    def failed(self):
+        return self._failed
+
+    def start(self):
+        assert self._start_time is None
+        self._start_time = time.time()
+
+    def stop(self):
+        assert self._start_time is not None and self._stop_time is None
+        self._stop_time = time.time()
+
+    def fail_with_error(self, ex: Exception):
+        assert self._fail_exception is None
+        self._fail_exception = ex
+        self._failed = True
+        self._fail_msgs.append(str(ex))
+
+    def fail_if(self, cond: bool, msg: str = ""):
+        if cond:
+            self._failed = True
+            self._fail_msgs.append(msg)
+
+    @property
+    def column_num(self):
+        return len(self.headers)
+
+    def append_row(self, *values: Any):
+        assert len(values) == len(self.headers)
+        self.rows.append(values)
+
+    def format(self):
```

Comment:
Can you give an example at here how the format will look like?

Reply:
The output is markdown formatted like this:

**[OTNS](https://github.com/openthread/ot-ns) Stress Tests Report Generated at 07/16 14:55:33**
#### Large Network Formation Test **Passed** _(07/16 06:53:51 ~ 07/16 06:55:33)_
| Simulation Time | Execution Time | Average Partition Count in 60s |
| :-: | :-: | :-: |
| 3600s | 101s | 15 |